pkg/parse/run.go

```diff
@@ -354,20 +361,22 @@ func parseSource(ctx context.Context, p Parser, trigger string, state *reconcile
 
 func parseAndUpdate(ctx context.Context, p Parser, trigger string, state *reconcilerState) status.MultiError {
 	sourceErrs := parseSource(ctx, p, trigger, state)
-	newSourceStatus := sourceStatus{
-		commit:     state.cache.source.commit,
-		errs:       sourceErrs,
-		lastUpdate: metav1.Now(),
-	}
-	if state.needToSetSourceStatus(newSourceStatus) {
-		if err := p.setSourceStatus(ctx, newSourceStatus); err != nil {
+	parseStatus := sourceStatus{
```

Comment:
I think I'm a bit confused as to what a sourceStatus is supposed to represent. Looks like here it's related to issues with parsing (i.e. marshalling?) but above it's related to issues reading some config files. Is it supposed to represent any error with reading from the filesystem at any stage of the pipeline?